### PR TITLE
4516654: Metalworks Demo: Window title not displayed fully in Low Vision Theme

### PR DESCRIPTION
**Pull request by prsadhuk**
In Metalworks demo in Low Vision theme, window title is not diplayed fully in this theme. The first letter of the title, that is letter "M", is cut vertically

image

It is because the iconsize is hardcoded to be 30 and windows titlePane title is rendered at same place irrespective of theme (only font size is made bigger from normal pt 12 to 24 for Low Vision theme) so icon width overlaps the title.
Icon width is made to be 24 in demo in sync with window title font size which made it work ok

image

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-4516654](https://bugs.openjdk.org/browse/JDK-4516654): Metalworks Demo: Window title not displayed fully in Low Vision Theme (**Bug** - P4)

### Reviewers
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14346/head:pull/14346` \
`$ git checkout pull/14346`

Update a local copy of the PR: \
`$ git checkout pull/14346` \
`$ git pull https://git.openjdk.org/jdk.git pull/14346/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14346`

View PR using the GUI difftool: \
`$ git pr show -t 14346`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14346.diff">https://git.openjdk.org/jdk/pull/14346.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14346#issuecomment-1580056721)

### CODE REVIEW

**Comment by TejeshR13** [comment]
Still slight overlap is seen with windows (Display scaling 150%).
MetalWork_24

**Review by TejeshR13 — COMMENTED** [review]
I applied the changes and observed that still some overlap is seen. Any additional properties settings are expected for this fix?

**Comment by DamonGuy** [comment]
> I applied the changes and observed that still some overlap is seen. Any additional properties settings are expected for this fix?

Applying the change worked for me. All I changed was the icon size, similar to this PR's changes, remade the jdk image, and reran Metalworks. Are you rebuilding the image after the change?

Screenshot 2023-06-07 at 11 55 59 AM

**Review by DamonGuy — COMMENTED** [review]
As with my previous comment, I am still getting the same cutoff issue as Tejesh. My personal device's setting has scaling to default but obvious the "Low Vision" theme has some additional scaling to adhere to the theme.

Maybe some additional changes are needed to fix this spacing?

**Comment by aivanov-jdk** [comment]
> My personal device's setting has scaling to default

And what is the default value?

Anything but 100% may cause rendering issues because the size of elements change and not always proportionally.

**Comment by DamonGuy** [comment]
> > My personal device's setting has scaling to default
> 
> And what is the default value?
> 
> Anything but 100% may cause rendering issues because the size of elements change and not always proportionally.

Sorry for the vagueness. It's at 100%

**Comment by prsadhuk** [comment]
The issue actually is caused by non-updation of `xOffSet` with correct value..
In MetalTitlePane, the initial `xOffset` is calculated with IMAGE_WIDTH value which is hardcoded to `16` so xOffset was 5+ 16 + 5 = 26 to draw the title string so we just about manage to draw the string outside the bounds of icon width which is `24 ` in `Metalworks Low Vision demo` but when icon image size is increased to 30 same xOffset of 26 is not sufficient so title string is drawn at xOffset 26 but [fillRect](https://github.com/openjdk/jdk/blob/6f492e800597c9ce332b9d5b54c00f551f145a0d/src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTitlePane.java#L852-L857) happens for width `30` so icon area fillect clears few pixels from string area and overlap of icon area and string happened.
Fix is to make sure `x Offset` is calculated with correct width so it will become 5 +30 (icon width) + 5 = 40

Issue of not getting large icon proportional to icon size is another issue and will be tackled separately.

**Review by TejeshR13 — APPROVED** [review]
Working as expected.

**Comment by aivanov-jdk** [comment]
> Fix is to make sure `x Offset` is calculated with correct width so it will become 5 +30 (icon width) + 5 = 40
> 
> Issue of not getting large icon proportional to icon size is another issue and will be tackled separately.

Looks good to me. No objections to handle following issues separately.

Metalworks with the fix

The icon in the titlebar isn't centered in Low Vision theme, however, JInternalFrame handles the situation correctly. The gap between the icon and text in the titlebar looks too large too.

**Review by aivanov-jdk — APPROVED** [review]

File: src/demo/share/jfc/Metalworks/BigContrastMetalTheme.java [hunk 1]
```diff
@@ -125,7 +125,7 @@ public void addCustomEntriesToTable(UIDefaults table) {
         table.put("TextField.border", textBorder);
         table.put("PasswordField.border", textBorder);
         table.put("TextArea.border", textBorder);
-        table.put("TextPane.font", textBorder);
+        table.put("TextPane.font", controlFont);
```
[the remark this answers is not in the document]
Comment (prsadhuk) [reply src/demo/share/jfc/Metalworks/BigContrastMetalTheme.java:128]: No, but it looks wrong passing border value for font

[the remark this answers is not in the document]
Comment (aivanov-jdk) [reply src/demo/share/jfc/Metalworks/BigContrastMetalTheme.java:128]: > No, but it looks wrong passing border value for font

I couldn't agree more. :)

**Pull request closed without merging**